pkg/rabbit/setup.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbit
+
+import (
+	"github.com/NeowayLabs/wabbit"
+	wabbitamqp "github.com/NeowayLabs/wabbit/amqp"
+	"go.uber.org/zap"
+)
+
+const maxRetries = 5
+
+func SetupRabbitMQ(
+	connection wabbit.Conn,
+	channel wabbit.Channel,
+	brokerURL string,
+	retryNumber *int,
+	logger *zap.SugaredLogger) {
+	if *retryNumber >= maxRetries {
+		logger.Fatalf("Max retries (%d) reached, unable to setup RabbitMQ correctly", maxRetries)
+		return
+	}
+
+	var err error
+	*retryNumber += 1
+	if connection == nil || connection.IsClosed() {
+		connection, err = wabbitamqp.Dial(brokerURL)
+		if err != nil {
+			logger.Fatalw("Failed to connect to RabbitMQ", zap.Error(err))
+		}
+	}
+
+	if channel, err = connection.Channel(); err != nil {
+		logger.Fatalw("Failed to open a channel", zap.Error(err))
+	}
+
+	// noWait is false
+	if err = channel.Confirm(false); err != nil {
+		logger.Fatalf("Failed to switch connection channel to confirm mode: %s", err)
+	}
+
+	if err != nil {
```

Comment:
this block may be some stale code? we have the go routine to attempt another reconnect. and the retryNumber += 0 doesn't do anything

Reply:
My bad that i left += 0 but this code makes sense im commenting for it to be clearer. Ill push after the tests are ready :)